simexpal/base.py

```diff
@@ -120,6 +120,12 @@ def __init__(self, basedir, yml):
 		self._variants = OrderedDict()
 		self._exp_infos = OrderedDict()
 
+		def check_key_value(yml_dict, target_key):
```

Comment:
I think this is rather inelegant: we assume here that `yml_dict` has a key containing the name. It's not clear to me that this is general enough in all cases.

Reply:
Also, the name of the function is not really self-explanatory. What about something like: `check_for_reserved_name()`?

Reply:
> Also, the name of the function is not really self-explanatory. What about something like: `check_for_reserved_name()`?

seems like a good name, done

Reply:
> I think this is rather inelegant: we assume here that `yml_dict` has a key containing the name. It's not clear to me that this is general enough in all cases.

I changed `check_for_reserved_name()` to only take a name as input and also added a print of the name if it does not pass the check